analyzers/src/SonarAnalyzer.CFG/ShimLayer/TypeDeclarationExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Lightup
+{
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    public static class TypeDeclarationExtensions
+    {
+        private static readonly Func<TypeDeclarationSyntax, ParameterListSyntax> ParameterListAccessor;
+        private static readonly Func<TypeDeclarationSyntax, ParameterListSyntax> RecordDeclarationParameterListAccessor;
+
+        static TypeDeclarationExtensions()
+        {
+            // TypeDeclarationSyntax.ParameterList was introduced in Roslyn 4.6 (C#12)
+            ParameterListAccessor = LightupHelpers.CreateSyntaxPropertyAccessor<TypeDeclarationSyntax, ParameterListSyntax>(typeof(TypeDeclarationSyntax), nameof(ParameterList));
+            // In earlier versions, the ParameterList was only available on the derived RecordDeclarationSyntax (starting from version 3.7)
+            // To work with version 3.7 to version 4.6 we need to special case the record declaration and access
+            // the parameter list from the derived RecordDeclarationSyntax.
+            if (SyntaxWrapperHelper.GetWrappedType(typeof(RecordDeclarationSyntaxWrapper)) is { } recordDeclaration)
```

Comment:
You are correct and the if is wrong. We need to remove the `if`. For versions <3.7 we also need to create the Accessor, but pass `null` for the type. That will enable a "Fallback" accessor which returns null instead of raising.

Reply:
Thanks @martin-strecker-sonarsource!